### PR TITLE
Fixup how -i, -n and --skip_config are used

### PR DESCRIPTION
**Pull request by npmitche**
How we were using -i (--ignore_platform), -n (--no_driver) and --skip_config was not actually doing what it was supposed to.

I believe this fixes it to work as described. 

Fixes https://github.com/chipsec/chipsec/issues/467

### CODE REVIEW

File: chipsec/chipset.py [hunk 1]
```diff
@@ -188,148 +188,137 @@ def detect_platform(self):
...
-                _unknown_pch = False
-                msg = f'PCH     : Actual values: VID = 0x{pch_vid:04X}, DID = 0x{pch_did:04X}, RID = 0x{pch_rid:02X}'
+        
+        if load_config:
+            if not ignore_platform and platform_code is None:
+                self.detect_platform()
+
+            elif platform_code in self.chipset_codes:
+                # Check if platform code passed in is valid and override configuration
+                msg = f'Platform: Actual values: VID = 0x{self.vid:04X}, DID = 0x{self.did:04X}, RID = 0x{self.rid:02X}'
```
[the remark this answers is not in the document]
Comment (npmitche) [reply chipsec/chipset.py:239]: This is when you pass in -p (or for the next block --pch) and it's capturing the actually read vid/did/rid and printing them, then re-assigning the vid/did/rid to match the platform/pch that as passed in.

[the remark this answers is not in the document]
Comment (fishilico) [reply chipsec/chipset.py:239]: Ok, that makes sense. Thanks for the explanation!

**Review by fishilico — APPROVED** [review]

**Comment by fishilico** [comment]
This seems to break commands not requiring a driver to be loaded.

Before (with `spi_rom.bin` being a SPI ROM dump written with `chipsec_util.py spi dump spi_rom.bin`):

```console
$ ./chipsec_util.py --helper linuxnativehelper spidesc spi_rom.bin
################################################################
##                                                            ##
##  CHIPSEC: Platform Hardware Security Assessment Framework  ##
##                                                            ##
################################################################
[CHIPSEC] Version  : 1.11.0
[CHIPSEC] Arguments: --helper linuxnativehelper spidesc 

usage: 
>>> chipsec_util spidesc <rom>
```

After this PR:

```console
$ ./chipsec_util.py --helper linuxnativehelper spidesc spi_rom.bin
################################################################
##                                                            ##
##  CHIPSEC: Platform Hardware Security Assessment Framework  ##
##                                                            ##
################################################################
[CHIPSEC] Version  : 1.11.0
[CHIPSEC] Arguments: --helper linuxnativehelper spidesc spi_rom.bin

ERROR: Platform is not supported ().
ERROR: To specify a cpu please use -p command-line option
ERROR: To specify a pch please use --pch command-line option

ERROR: If the correct configuration is not loaded, results should not be trusted.

$ ./chipsec_util.py --helper linuxnativehelper -i spidesc spi_rom.bin
################################################################
##                                                            ##
##  CHIPSEC: Platform Hardware Security Assessment Framework  ##
##                                                            ##
################################################################
[CHIPSEC] Version  : 1.11.0
[CHIPSEC] Arguments: --helper linuxnativehelper -i spidesc spi_rom.bin

WARNING: Platform dependent functionality is likely to be incorrect
[!]       Unknown Platform: VID = 0xFFFF, DID = 0xFFFF, RID = 0xFF; Using Default.
[!]       Unknown PCH: VID = 0xFFFF, DID = 0xFFFF, RID = 0xFF; Using Default.
WARNING: [!]            If using configuration data, results from this system will likely be incorrect.
Traceback (most recent call last):
  File "/git/chipsec/./chipsec_util.py", line 208, in <module>
    sys.exit(main())
  File "/git/chipsec/./chipsec_util.py", line 203, in main
    return chipsecMain.main()
  File "/git/chipsec/./chipsec_util.py", line 183, in main
    print_banner_properties(self._cs, os_version())
  File "/git/chipsec/chipsec/banner.py", line 85, in print_banner_properties
    logger().log(chipsec_banner_properties(cs, os_version))
  File "/git/chipsec/chipsec/banner.py", line 67, in chipsec_banner_properties
    [CHIPSEC]    CPUID: {cs.get_cpuid()}
  File "/git/chipsec/chipsec/chipset.py", line 192, in get_cpuid
    (eax, _, _, _) = self.cpu.cpuid(0x01, 0x00)
  File "/git/chipsec/chipsec/hal/cpu.py", line 60, in cpuid
    (eax, ebx, ecx, edx) = self.helper.cpuid(eax, ecx)
  File "/git/chipsec/chipsec/helper/basehelper.py", line 179, in cpuid
    raise NotImplementedError()
NotImplementedError
```

**Comment by npmitche** [comment]
I thought I tried all the permutations, but did not try passing in a helper. Let me look into it.

**Comment by npmitche** [comment]
What I've discovered:
Some of the changes make it so when a chipsec_uitil cmd returns `False` for `requires_driver()`, the driver is not loaded. With no driver loaded, it is not able to identify the VID, which chipsec uses to load configurations (even default/common configs). So, if you passed in anything for the platform, eg `-p ivb`  it should work (assuming you have #1817). 

This is a weird case because spidesc relies on register definitions in a config but it doesn't actually rely on the driver. A simple solution is to simply require the driver for that command, or change the way you call the module to be sure to pass in the `-p` flag. 

Thoughts, comments? 

@frinzell Do you have any thoughts on this?

**Comment by npmitche** [comment]
I believe this PR is working as indented. In future work, I will work on splitting out the  `requires_driver` function into 3: `requires_driver() -> bool`, `requires_config() -> bool`, and `parse_arguments() -> None`. This will be a slightly heavier lift so not sure when I'll get around to it, but I think that makes the most sense.

**Review by dscott90 — APPROVED** [review]

**Comment by npmitche** [comment]
Rebased to resolve merge conflicts.

**Review by Sae86 — APPROVED** [review]

**Pull request merged**